### ISSUE TITLE
System BootOrder not found (UEFI)

### ISSUE THREAD

**Issue by peter-lxc** [oamg/convert2rhel#180]
I had centos 8 using lvm and i ran the convert2rhel on a physical machine and now my system doesnt boot have an error on boot 

"System BootOrder not found. Initializing defaults"
Creating boot entry Boot003 with label "Red Hat Enterprise Linux£ for file '\EFI\REDHAT\shimx64.efi"

Any ideas ???

**Comment by bocekm** [comment]
Hi @peter-lxc, I'm sorry you hit that. It seems your system is UEFI-based and convert2rhel currently does not handle that. I handles just BIOS-based systems. I'll make sure we add a check to not allow the conversions if we detect UEFI to prevent these issues.

**Comment by dill-wood** [comment]
Unfortunate. I will not be able to use the convert tool until UEFI is supported. 

Is this on a roadmap to be supported potentially before the switch to CentOS stream from 8?

Thank you,

**Comment by bocekm** [comment]
@dill-wood, yes, it is high on our todo list.

**Comment by ZhukovGreen** [comment]
@dill-wood We'll notify you when this feature will be added

**Comment by godofgrunts** [comment]
I just ran into this problem today. Seeing as this issue is almost a year old now, is this a high priority concern?

Is it possible to configure efibootmgr manually afterwards?

**Comment by SpyTec** [comment]
@godofgrunts This has been fixed since `convert2rhel-0.24-2`, we just forgot to update this issue. 

It could be you're running into [this bug](https://bugzilla.redhat.com/show_bug.cgi?id=2027686), which we have documented in [known issues and limitations](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/converting_from_an_rpm-based_linux_distribution_to_rhel/index#ref_known-issues-and-limitations_assembly_troubleshooting-rhel-conversions) section. 

If that is not the issue, what version are you running? Could you provide some logs?

**Comment by godofgrunts** [comment]
This was a VM on a production box so I already reverted it from a snapshot.

I'm going to attempt this again on Sunday during a maintenance window
though so I'll make sure to get the logs before I try to reboot it.

I was using whatever binary was in the repo the instructions tell you to
add, but I was looking though the convert2rhel in /bin and I believe it
said it required convert2rhel=0.25 if that helps.

**Comment by bampolsk** [comment]
with convert2rhel 0.25 it's possible to migrate UEFI VMs, but only after turning off SecureBoot.

**Comment by Githopp192** [comment]
i'on a HPE DL 20 .. Secure Boot Status is disabled .. run RHEL convert2rhel Conversion from Centos 8 to RHEL 8.5.

dnf info installed | grep -i convert2rhe
Name         : convert2rhel
Source       : convert2rhel-0.25-1.el8.src.rpm
From repo    : convert2rhel-for-rhel-8-rpms


[03/31/2022 19:37:05] TASK - [Convert: Configure the bootloader] ********************************
UEFI binary found: /boot/efi/EFI/redhat/shimx64.efi
Copying GRUB2 configuration files to the new UEFI directory /boot/efi/EFI/redhat/.
Copying '/boot/efi/EFI/centos/grubenv' to '/boot/efi/EFI/redhat/grubenv'
Copying '/boot/efi/EFI/centos/grub.cfg' to '/boot/efi/EFI/redhat/grub.cfg'
WARNING - The folder /boot/efi/EFI/centos/ is left untouched. You may remove the folder manually after you ensure there is no custom data you would need.
Loading the bootloader configuration.
Adding a new UEFI bootloader entry for RHEL.
Adding 'Red Hat Enterprise Linux 8' UEFI bootloader entry.
Verifying the new UEFI bootloader entry.
CRITICAL - Unable to find the new UEFI bootloader entry.
The migration of the bootloader setup was not successful.
Do not reboot your machine before doing a manual check of the
bootloader configuration. Ensure that grubenv and grub.cfg files
are present in the /boot/efi/EFI/redhat/ directory and that
a new bootloader entry for Red Hat Enterprise Linux exists
(check `efibootmgr -v` output).
The entry should point to '\EFI\redhat\shimx64.efi'.
WARNING - Conversion process interrupted and manual user intervention will be necessary.

**Comment by Githopp192** [comment]
did reboot the system & and fortunately the system comes up - but before doing that - as precaution i did:

> grub2-mkconfig -o /boot/efi/EFI/redhat/grub.cfg

Generating grub configuration file ...
Adding boot menu entry for EFI firmware configuration
done

> efibootmgr

Red Hat entry does exist, but did remove the CENTOS entires manually: 

> efibootmgr -Bb 0016 .... 18

**Comment by LuckyDams** [comment]
Hi,
Had similar issue with conversion from Rocky 8.6 with EFI boot using convert2rhel 0.25 (ending to grub prompt after first reboot). 

Command `grub2-mkconfig -o /boot/efi/EFI/redhat/grub.cfg` was missing leading to broken symbolic link to `/etc/grub2-efi.cfg`. 
However `efibootmgr` was managed correctly by the script to create new entry & remove old ones.

Here the related ouptut:

```
[07/26/2022 22:59:23] TASK - [Convert: Configure the bootloader] ********************************
UEFI binary found: /boot/efi/EFI/redhat/shimx64.efi
[07/26/2022 22:59:23] FILE - Skipping copying GRUB files - only related to CentOS Linux.
Loading the bootloader configuration.
[07/26/2022 22:59:23] FILE - Calling command '/usr/sbin/efibootmgr -v'
[07/26/2022 22:59:23] FILE - Bootloader setup:
Current boot: 0004
Boot order: 0004, 0000, 0001, 0002, 0003, 0005
Boot entries:
- 0000: EFI Virtual disk (0.0)
- 0001: EFI VMware Virtual SATA CDROM Drive (0.0)
- 0002: EFI Network
- 0003: EFI Internal Shell (Unsupported option)
- 0004: Rocky Linux
- 0005: EFI Virtual disk (1.0)
Adding a new UEFI bootloader entry for RHEL.
[07/26/2022 22:59:23] FILE - Calling command '/usr/sbin/grub2-probe --target=device /boot/efi/'
[07/26/2022 22:59:23] FILE - Calling command 'lsblk -spnlo MAJ:MIN /dev/sda1'
[07/26/2022 22:59:23] FILE - Calling command '/usr/sbin/grub2-probe --target=device /boot/efi/'
[07/26/2022 22:59:23] FILE - Calling command 'lsblk -spnlo name /dev/sda1'
[07/26/2022 22:59:23] FILE - Block device: /dev/sda
[07/26/2022 22:59:23] FILE - ESP device number: {'major': 8, 'minor': 1}
[07/26/2022 22:59:23] FILE - The new UEFI binary: /boot/efi/EFI/redhat/shimx64.efi
Adding 'Red Hat Enterprise Linux 8' UEFI bootloader entry.
[07/26/2022 22:59:23] FILE - Calling command '/usr/sbin/efibootmgr --create --disk /dev/sda --part 1 --loader \EFI\redhat\shimx64.efi --label Red Hat Enterprise Linux 8'
Verifying the new UEFI bootloader entry.
[07/26/2022 22:59:23] FILE - Calling command '/usr/sbin/efibootmgr -v'
[07/26/2022 22:59:23] FILE - Bootloader setup:
Current boot: 0004
Boot order: 0006, 0004, 0000, 0001, 0002, 0003, 0005
Boot entries:
- 0000: EFI Virtual disk (0.0)
- 0001: EFI VMware Virtual SATA CDROM Drive (0.0)
- 0002: EFI Network
- 0003: EFI Internal Shell (Unsupported option)
- 0004: Rocky Linux
- 0005: EFI Virtual disk (1.0)
- 0006: Red Hat Enterprise Linux 8
The 'Red Hat Enterprise Linux 8' bootloader entry has been added.
Removing the original UEFI bootloader entry.
[07/26/2022 22:59:23] FILE - Calling command '/usr/sbin/efibootmgr -Bb 0004'
The removal of the original, currenly used UEFI bootloader entry '0004' (Rocky Linux) has been successful.

```

**Comment by r0x0d** [comment]
Hi, @LuckyDams! Sorry that you had problems with the conversion on C2R 0.25. 

Luckily, a fix for that is already in upstream: https://github.com/oamg/convert2rhel/pull/417. It should be available in the next release (0.27).

**Comment by bocekm** [comment]
Fixed delivered as part of the release 1.0: https://github.com/oamg/convert2rhel/releases/tag/v1.0.
